app/views/devise/confirmations/new.html.erb

```diff
@@ -5,7 +5,7 @@
 
   <div class="field">
     <%= f.label :email %><br />
-    <%= f.email_field :email, autofocus: true %>
+    <%= f.email_field :email, required: true, readonly: true, input_html: { value: (resource.pending_reconfirmation? ? resource.unconfirmed_email : resource.email) } %> 
```

Comment:
That seems to work as well, but see my notes in the commit description on why :autofocus should be removed, and :readonly added. I also presume :required is nice since it will HTML5 validate the form client-side, so that it's sure to contain an email before submitting.